### DISCUSSION TITLE
On the GPL-3.0 license

### DISCUSSION THREAD

**Discussion by FabianIsensee** [seung-lab/connected-components-3d#120] [General]
Hey there,
awesome package that really makes working with connected components in 3D a breeze! We would love to use it in some of our python packages but are worried about the GLP-3.0 license as we are following a more permissive licensing policy (Apache 2.0, MIT and such) that we would like to keep moving forward. We are a German tax funded research institution but our open source packages are used by a variety of users, including industry. 
Do you consider python packages that import and use this repository -but do not extend its functionality- derivative work that must also be licensed under GPL-3.0? 
Best,
Fabian

**Comment by william-silversmith** [comment]
Hi Fabian,

A straightforward interpretation of the GPL3 is that software that links to GPL'd software is covered under GPL. `import` is the Python command for linking libraries within the Python language.

I might consider shifting cc3d to LGPL though.

**Reply by FabianIsensee** [reply]
Hey @william-silversmith,
thank you for your prompt response. LGPL would enable us to link with cc3d without having to license our own work under GPL, which sounds great. Could you please update me on the current status of your considerations regarding LGPL? Is a change likely to happen soon, or is it still in the exploratory phase?
Thanks :pray:,
Fabian

**Reply by william-silversmith** [reply]
Okay, I changed it to LGPLv3.

**Reply by FabianIsensee** [reply]
Thank you so much! Awesome!
